packages/react/src/Header/Header.tsx

```diff
@@ -11,54 +11,49 @@ import { Logo } from '../Logo'
 import type { LogoBrand } from '../Logo'
 
 export type HeaderProps = {
-  /** The name of the application. */
+  /** A site-wide title for the website or application. */
   appName?: string
-  /** The list of menu links. Use a Page Menu here. */
-  links?: ReactNode
-  /** The name of the brand for which to display the logo. */
+  /** The HTML element to use. */
+  as?: 'header' | 'div'
+  /** The brand for which to display the logo. */
   logoBrand?: LogoBrand
   /** The url for the link on the logo. */
   logoLink?: string
   /** The accessible text for the link on the logo. */
   logoLinkTitle?: string
-  /** A button to toggle the visibility of a Mega Menu. */
+  /** A secondary menu of links and buttons. Use a Page Menu. Can include buttons to show the Mega Menu or a Search Field. */
   menu?: ReactNode
-} & HTMLAttributes<HTMLElement>
+} & HTMLAttributes<HTMLDivElement>
 
 export const Header = forwardRef(
   (
     {
       appName,
+      as: Tag = 'div',
       className,
-      links,
       logoBrand = 'amsterdam',
       logoLink = '/',
       logoLinkTitle = 'Ga naar de homepage',
       menu,
       ...restProps
     }: HeaderProps,
-    ref: ForwardedRef<HTMLElement>,
-  ) => {
-    return (
-      <>
-        <header {...restProps} ref={ref} className={clsx('ams-header', className)}>
-          <a href={logoLink} className="ams-header__logo-link">
-            <span className="ams-visually-hidden">{logoLinkTitle}</span>
-            <Logo brand={logoBrand} />
-          </a>
-          {links && <div className="ams-header__links">{links}</div>}
-          {menu && <div className="ams-header__menu">{menu}</div>}
-          {appName && (
-            <div className="ams-header__app-name">
-              <Heading level={1} size="level-3" className="ams-header__app-name-heading">
-                {appName}
-              </Heading>
-            </div>
-          )}
-        </header>
-      </>
-    )
-  },
+    ref: ForwardedRef<HTMLDivElement>,
+  ) => (
+    <Tag {...restProps} ref={ref} className={clsx('ams-header', className)}>
+      <div className="ams-header__section">
```

Comment:
Do you really need a wrapping div? You could give the logo, app-name and menu their own flex properties to achieve the same effect. Perhaps some `display: contents` magic?

Reply:
It seems more robust to encode the two sections that the header clearly has. I tried having logo, app name and page menu as direct children of the menu and concluded:

- This allows the app name to wrap below the logo instead of wrapping its text
- It requires left margins on the app name and the page menu, which couples whitespace to semantic elements. Coupling it to structural sections allows for different elements while keeping spacing intact.

Reply:
I understand, in my previous attempts I solved this with grid areas.